src/App.tsx

```diff
@@ -1,61 +1,117 @@
+import { useState } from 'react';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+import { Todo, User } from './types';
+import { TodoList } from './components/TodoList/TodoList';
+
+function getUser(userId: number): User | null {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  return foundUser || null;
+}
+
+export const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
+
+const visibleTodos = [...todos];
 
 export const App = () => {
+  const [title, setTitle] = useState('');
+  const [userId, setUserId] = useState(0);
+  const [errorTittle, setErrorTittle] = useState(false);
+  const [errorUser, setErrorUser] = useState(false);
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (userId === 0 || title === '') {
+      setErrorTittle(title === '');
+      setErrorUser(userId === 0);
+
+      return;
+    }
+
+    const maxId = Math.max(...visibleTodos.map((todo) => todo.id));
+
+    const newTodo = {
+      id: maxId + 1,
+      title,
+      userId,
+      completed: false,
+      user: getUser(userId),
+    };
+
+    visibleTodos.push(newTodo);
+
+    setTitle('');
+    setUserId(0);
+
+    setErrorUser(false);
+    setErrorTittle(false);
```

Comment:
```suggestion
    setErrorUser(false);
    setErrorTittle(false);
```
I feel like your code gonna work even without these 2 lines of code, because you already handled cases if one of them is invalid